### PR TITLE
Remove all secrets on delete incl. pooler

### PR DESCRIPTION
**Pull request by FxKu**
This PR sets secrets on SYNC, so that they can be deleted.

On connection pooler deletion the corresponding secret is now removed via API call. Because of the way roles and secrets are synced with the `needCoonectionPooler` condition, the pooler user and secret are no longer set after deletion. But resources would still be there. For the database user it's fine - operator does not delete those in general - but the secret should go.

### CODE REVIEW

File: pkg/cluster/resources.go [hunk 1]
```diff
@@ -221,6 +219,23 @@ func (c *Cluster) deleteConnectionPooler() (err error) {
...
+	secret, err := c.KubeClient.
+		Secrets(c.Namespace).
+		Get(context.TODO(), secretName, metav1.GetOptions{})
+
+	if err != nil {
+		c.logger.Debugf("could not get connection pooler secret %q: %v", secretName, err)
+	} else {
+		uid := secret.UID
+		if err = c.deleteSecret(uid, *secret); err != nil {
+			return fmt.Errorf("could not delete pooler secret: %v", err)
```
Comment (Jan-M) [inline pkg/cluster/resources.go:234]: does this prevent deletion of e.g deployment? disable pooler should cleanup as good as it gets.

**Comment by Jan-M** [comment]
:+1:

**Comment by Jan-M** [comment]
:+1:

File: pkg/cluster/resources.go (hunk 1 above)
Comment (FxKu) [reply pkg/cluster/resources.go:233]: nope. 1. Deployment, 2. Service, 3. Secret

It's true, however, that if 1. or 2. return an error the rest is not deleted.

**Comment by FxKu** [comment]
:+1:

**Pull request merged**